main_server/main.py

```diff
@@ -146,6 +146,27 @@ def do_POST(self):
                 _rfile, self.headers, json_rfile, repo, originRepo))
             calc.start()
 
+    #GET for UTR checks
+    def do_GET(self):
+        self.useragent = self.headers.get("User-Agent")
+        match = re.search(r"(?:UptimeRobot)", self.useragent, flags=re.MULTILINE|re.IGNORECASE)
+        if match:
+            self.send_response(200, "OK, Test recieved!")
+            self.end_headers()
+            print("GET-Test received, sent 200")
+        else:
+            self.send_response(403, "Forbidden")
+
+    #HEAD for UTR checks
+    def do_HEAD(self):
+        self.useragent = self.headers.get("User-Agent")
```

Comment:
User-Agent is bad to detect the services ... use a IP whitelist instead

Reply:
Added